command/cliconfig/cliconfig.go

```diff
@@ -91,12 +97,23 @@ func LoadConfig() (*Config, tfdiags.Diagnostics) {
 		}
 	}
 
-	if configDir, err := ConfigDir(); err == nil {
-		if info, err := os.Stat(configDir); err == nil && info.IsDir() {
-			dirConfig, dirDiags := loadConfigDir(configDir)
-			diags = diags.Append(dirDiags)
-			config = config.Merge(dirConfig)
+	// Unless the user has specifically overridden the configuration file
+	// location using an environment variable, we'll also load what we find
+	// in the config directory. We skip the config directory when source
+	// file override is set because we interpret the environment variable
+	// being set as an intention to ignore the default set of CLI config
+	// files because we're doing something special, like running Terraform
+	// in automation with a locally-customized configuration.
+	if cliConfigFileOverride() == "" {
+		if configDir, err := ConfigDir(); err == nil {
+			if info, err := os.Stat(configDir); err == nil && info.IsDir() {
+				dirConfig, dirDiags := loadConfigDir(configDir)
+				diags = diags.Append(dirDiags)
+				config = config.Merge(dirConfig)
+			}
 		}
+	} else {
+		log.Printf("Not reading CLI config directory because config location is overridden by environment variable")
```

Comment:
Does this need a log level?  `"[INFO]"` (or whatever you deem appropriate) at the start of the string?

Reply:
It seems like some other log lines in this area don't tend to have it, but I suppose I don't really know why that is, so I will add levels to all the new ones I've added around here.